From 366ea72e9d8d73698d6c87f7e515290b1fd3f062 Mon Sep 17 00:00:00 2001
From: Peng Zhang <zhangpeng.00@bytedance.com>
Date: Tue, 25 Apr 2023 19:05:07 +0800
Subject: [PATCH 5/9] maple_tree: Remove an if statement that cannot be true

Because the commit 06e8fd999334b ("maple_tree: fix mas_empty_area() search")
is merged, this if statement cannot be true, so delete it.

Signed-off-by: Peng Zhang <zhangpeng.00@bytedance.com>
---
 lib/maple_tree.c | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/lib/maple_tree.c b/lib/maple_tree.c
index 374f0cd56..818e92a21 100644
--- a/lib/maple_tree.c
+++ b/lib/maple_tree.c
@@ -5108,9 +5108,6 @@ static inline bool mas_anode_descend(struct ma_state *mas, unsigned long size)
 			return true;
 		}
 	}
-
-	if (mte_is_root(mas->node))
-		found = true;
 done:
 	mas->offset = offset;
 	return found;
-- 
2.40.0.71.g950264636c

